### PR TITLE
[FLOC-3672] Run agent tests as root.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-3672

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -179,26 +179,7 @@ common_cli:
...
-    if is_centos || is_ubuntu; then
-      # some of our tests (which run as root), leave some files behind when
-      # they abort or fail to complete.
-      # We use 'sudo' so that we can remove those files before a next job run.
-      sudo rm -rf \${venv}
-      sudo rm -f results.xml
-      sudo rm -f trial.log
-      sudo rm -rf _trial_temp/
-    fi
+    rm -rf \${venv}
```
Comment (tomprince) [inline build.yaml]: This always occurs after running `git clean -fxd` so there will never be extraneous files in the repository. Thus we only need to clean files that are outside of it.

File: flocker/node/agents/test/test_blockdevice_manager.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ def setUp(self):
         """
         Establish testing infrastructure for test cases.
         """
+        super(BlockDeviceManager, self)
```
Comment (tomprince) [inline flocker/node/agents/test/test_blockdevice_manager.py]: These tests never ran, so this failure was never noticed.

File: flocker/node/agents/test/test_blockdevice_manager.py [hunk 3]
```diff
@@ -34,6 +34,7 @@ def setUp(self):
         """
         Establish testing infrastructure for test cases.
         """
+        super(BlockDeviceManagerTests, self).setUp()
```
Comment (tomprince) [inline flocker/node/agents/test/test_blockdevice_manager.py]: These tests never ran, so the fact that this didn't upcall was never noticed.

**Comment by jml** [comment]
:+1: 

Merge when OK w/ test results.

**Pull request merged**